### PR TITLE
maintenance: fix doc generation

### PR DESCRIPTION
**Pull request by psafont**
Doc building was broken using odoc, changing to ocamldoc tags fixes it.

The documentation was manually checked, same as https://github.com/xapi-project/xcp-idl/pull/300

### CODE REVIEW

File: generator/lib/data.ml [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+    VM uuid or a Xenstore path or anything else chosen by the toolstack.
+    Implementations should not assume the string has any meaning. *)
 type domain = string
 [@@deriving rpcty]
-[@@doc [
-  "A string representing a Xen domain on the local host. The string is";
-  "guaranteed to be unique per-domain but it is not guaranteed to take";
-  "any particular form. It may (for example) be a Xen domain id, a Xen";
-  "VM uuid or a Xenstore path or anything else chosen by the toolstack.";
-  "Implementations should not assume the string has any meaning."]]
```
Comment (lippirk) [inline generator/lib/data.ml]: i was under the impression that these tags generate docs for the IDL? but i could be completely wrong

File: generator/lib/common.ml [hunk 2]
```diff
@@ -17,14 +17,15 @@ let dbg = Param.mk ~name:"dbg" ~description:["Debug context from the caller"] Ty
...
+type uri = string [@@deriving rpcty]
 
+(** List of blocks for copying. *)
 type blocklist = {
-  blocksize : int
-      [@doc ["size of the individual blocks"]];
-  ranges : (int64 * int64) list
-      [@doc ["list of block ranges, where a range is a (start,length) pair, measured in units of [blocksize]"]];
-} [@@deriving rpcty] [@@doc ["List of blocks for copying"]]
+  blocksize : int ; (** Size of the individual blocks. *)
```
Comment (lippirk) [inline generator/lib/common.ml]: `int ;` -> `int;`

Comment (lippirk) [reply generator/lib/common.ml]: oh, you have done this everywhere, is that the convention?

Comment (psafont) [reply generator/lib/common.ml]: I followed the notation in https://caml.inria.fr/pub/docs/manual-ocaml/ocamldoc.html#sec359

Comment (lindig) [reply generator/lib/common.ml]: I, too, think a space before a semicolon is unconventional (both in source code and in prose) and was surprised to see it in the manual. But `ocamlformat` also formats code in this way so there seems to be some agreement here.

**Pull request merged**